packages/backend/src/invoice/routes.test.ts

```diff
@@ -0,0 +1,322 @@
+import * as httpMocks from 'node-mocks-http'
+import Knex from 'knex'
+import { WorkerUtils, makeWorkerUtils } from 'graphile-worker'
+import { v4 as uuid } from 'uuid'
+
+import { createContext } from '../tests/context'
+import { PaymentPointerService } from '../payment_pointer/service'
+import { PaymentPointer } from '../payment_pointer/model'
+import { createTestApp, TestContainer } from '../tests/app'
+import { resetGraphileDb } from '../tests/graphileDb'
+import { GraphileProducer } from '../messaging/graphileProducer'
+import { Config, IAppConfig } from '../config/app'
+import { IocContract } from '@adonisjs/fold'
+import { initIocContainer } from '../'
+import { AppServices } from '../app'
+import { truncateTables } from '../tests/tableManager'
+import { randomAsset } from '../tests/asset'
+import { InvoiceService } from './service'
+import { Invoice } from './model'
+import { InvoiceRoutes, MAX_EXPIRY } from './routes'
+import { AppContext } from '../app'
+
+describe('Invoice Routes', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let knex: Knex
+  let workerUtils: WorkerUtils
+  let paymentPointerService: PaymentPointerService
+  let invoiceService: InvoiceService
+  let config: IAppConfig
+  let invoiceRoutes: InvoiceRoutes
+  const messageProducer = new GraphileProducer()
+  const mockMessageProducer = {
+    send: jest.fn()
+  }
+
+  beforeAll(
+    async (): Promise<void> => {
+      config = Config
+      config.publicHost = 'https://wallet.example'
+      deps = await initIocContainer(config)
+      deps.bind('messageProducer', async () => mockMessageProducer)
+      appContainer = await createTestApp(deps)
+      workerUtils = await makeWorkerUtils({
+        connectionString: appContainer.connectionUrl
+      })
+      await workerUtils.migrate()
+      messageProducer.setUtils(workerUtils)
+      knex = await deps.use('knex')
+    }
+  )
+
+  let asset: { code: string; scale: number }
+  let paymentPointer: PaymentPointer
+  let invoice: Invoice
+
+  beforeEach(
+    async (): Promise<void> => {
+      paymentPointerService = await deps.use('paymentPointerService')
+      invoiceService = await deps.use('invoiceService')
+      config = await deps.use('config')
+      invoiceRoutes = await deps.use('invoiceRoutes')
+
+      asset = randomAsset()
+      paymentPointer = await paymentPointerService.create({ asset })
+      invoice = await invoiceService.create({
+        paymentPointerId: paymentPointer.id,
+        description: 'text',
+        expiresAt: new Date(Date.now() + 30_000),
+        amountToReceive: BigInt(123)
+      })
+    }
+  )
+
+  afterEach(
+    async (): Promise<void> => {
+      await truncateTables(knex)
+    }
+  )
+
+  afterAll(
+    async (): Promise<void> => {
+      await resetGraphileDb(knex)
+      await appContainer.shutdown()
+      await workerUtils.release()
+    }
+  )
+
+  describe('get', (): void => {
```

Comment:
You could also validate the response by doing a `pay.setupPayment` with the invoice url, but it looks like pay would need some fixes
https://github.com/interledgerjs/interledgerjs/issues/257